docs/DeveloperGuide.md

```diff
@@ -184,13 +184,46 @@ A `TableNumber` object stores a table number as an integer. It is wrapped in an
   * Pros: Allows for greater flexibility in implementing features that are specific to either guests or vendors.
   * Cons: More code duplication.
 
+### Add feature
+
+#### Implementation
+
+The add feature allows users to add new guests or vendors with the compulsory field `Name`, along with any of the optional
+fields mentioned in the [Tracking of Guests and Vendors](#tracking-of-guests-and-vendors) section. The feature is implemented through the
+classes `GuestAddCommand` and `VendorAddCommand`. The implementation of the various classes facilitating the add feature
+on `Guest` and `Vendor` objects differ only in specifics that are not relevant here, so the keywords `Guest` and `Vendor` will be 
+replaced by `XYZ` (e.g. `XYZAddCommand` can be substituted with both `GuestAddCommand` and `VendorAddCommand`).
+
+Given below is an example usage scenario and how the add mechanism behaves at each step. You may also refer to the sequence
+diagrams provided for a visual representation of the process.
+
+Step 1. The user launches the application. WedLog shows all guests and vendors in their respective lists.
+
+Step 2. The user executes `xyz add n/Annette t/friend`, where `xyz` is either `guest` or `vendor`. This allows the user to add
+a guest or vendor with the name `Annette` and tag `friend`. 
+
+Step 3. `AddressBookParser` parses the `xyz` keyword and creates the `XYZCommandParser`. 
+`XYZCommandParser` parses the `add` keyword and creates a `XYZAddCommandParser` object. It also calls `XYZAddCommandParser#parse` to parse the inputted fields.
+
+<puml src="diagrams/AddParseSequenceDiagram.puml" alt="AddParseSequenceDiagram" />
+
+Step 4. `#parse` calls upon `ParserUtil#parseABC`, where `ABC` is the field being added, to check the validity of the
+user input and convert it into field objects (e.g. string representing a new name into a `Name` object).
+`#parse` then creates an `XYZAddCommand` using the created field objects.
+
+<puml src="diagrams/AddParseSequenceDiagramRef.puml" alt="AddParseSequenceDiagramRef" />
+
+Step 5. Lastly, `XYZAddCommand#execute` adds a `XYZ` with the given values to the `XYZGuestList`.
```

Comment:
```suggestion
Step 5. Lastly, `XYZAddCommand#execute` adds a `XYZ` with the given values to the `UniqueXYZList`.
```

Reply:
I think you might have been referring to the UniqueGuestList/ UniqueVendorList

Reply:
Yup I did, thanks for the spot!